build/building/packaging.md

```diff
@@ -113,122 +127,149 @@ your Dockerfile, and should be the first line in Dockerfiles.
 > release of the version 1 syntax. BuildKit automatically checks for updates of
 > the syntax before building, making sure you are using the most current version.
 
-Next we define the first instruction:
+### Base image
+
+The line following the syntax directive defines what base image to use:
 
 ```dockerfile
 FROM ubuntu:22.04
 ```
 
-Here the [`FROM` instruction](../../engine/reference/builder.md#from) sets our
-base image to the 22.04 release of Ubuntu. All following instructions are
-executed on this base image, in this case, an Ubuntu environment. The notation
-`ubuntu:22.04`, follows the `name:tag` standard for naming docker images. When
-you build your image you use this notation to name your images and use it to
-specify any existing Docker image. There are many public images you can
-leverage in your projects. Explore [Docker Hub](https://hub.docker.com/search?image_filter=official&q=&type=image){:target="blank" rel="noopener" class=""}
-to find out.
+The [`FROM` instruction](../../engine/reference/builder.md#from) sets your base
+image to the 22.04 release of Ubuntu. All instructions that follow are executed
+in this base image: an Ubuntu environment. The notation `ubuntu:22.04`, follows
+the `name:tag` standard for naming Docker images. When you build images, you
+use this notation to name your images. There are many public images you can
+leverage in your projects, by importing them into your build steps using the
+Dockerfile `FROM` instruction.
+
+[Docker Hub](https://hub.docker.com/search?image_filter=official&q=&type=image)
+contains a large set of official images that you can use for this purpose.
+
+### Environment setup
+
+The following line executes a build command inside the base image.
 
 ```dockerfile
 # install app dependencies
 RUN apt-get update && apt-get install -y python3 python3-pip
 ```
 
-This [`RUN` instruction](../../engine/reference/builder.md#run) executes a shell
-command in the [build context](context.md).
+This [`RUN` instruction](../../engine/reference/builder.md#run) executes a
+shell in Ubuntu that updates the APT package index and installs Python tools in
+the container.
 
-In this example, our context is a full Ubuntu operating system, so we have
-access to its package manager, apt. The provided commands update our package
-lists and then, after that succeeds, installs `python3` and `pip`, the package
-manager for Python.
+### Comments
 
-Also note `# install app dependencies` line. This is a comment. Comments in
+Note the `# install app dependencies` line. This is a comment. Comments in
 Dockerfiles begin with the `#` symbol. As your Dockerfile evolves, comments can
-be instrumental to document how your dockerfile works for any future readers
-and editors of the file.
+be instrumental to document how your Dockerfile works for any future readers
+and editors of the file, including your future self!
```

Comment:
a controversial '!'

Reply:
Yep, this is somewhat tongue in cheek so I think a `!` is good here :-)